### PR TITLE
[MM-44573] Sanitize options

### PR DESCRIPTION
**Pull request by Willyfrog**
#### Summary

Sanitize the user object as defined by the config.

#### Ticket Link

[MM-44573](https://mattermost.atlassian.net/browse/MM-44573)

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: store/localcachelayer/main_test.go [hunk 1]
```diff
@@ -104,8 +104,8 @@ func getMockStore() *mocks.Store {
 	fakePosts := &model.PostList{}
 	fakeOptions := model.GetPostsOptions{ChannelId: "123", PerPage: 30}
 	mockPostStore := mocks.PostStore{}
-	mockPostStore.On("GetPosts", fakeOptions, true).Return(fakePosts, nil)
-	mockPostStore.On("GetPosts", fakeOptions, false).Return(fakePosts, nil)
+	mockPostStore.On("GetPosts", fakeOptions, true, map[string]bool{}).Return(fakePosts, nil)
+	mockPostStore.On("GetPosts", fakeOptions, false, map[string]bool{}).Return(fakePosts, nil)
```
Comment (koox00) [inline store/localcachelayer/main_test.go:108]: not sure if this file is auto generated by codegen, but I had to manually edit these 2 lines for the tests to pass.

**Review by koox00 — APPROVED** [review]
💯

**Comment by amyblais** [comment]
@mkraft Please help prioritize this PR review.

File: app/post.go [hunk 2]
```diff
@@ -965,10 +965,11 @@ func (a *App) GetPostsAfterPost(options model.GetPostsOptions) (*model.PostList,
 func (a *App) GetPostsAroundPost(before bool, options model.GetPostsOptions) (*model.PostList, *model.AppError) {
 	var postList *model.PostList
 	var err error
+	sanitize := a.Config().GetSanitizeOptions()
```
Comment (jwilander) [inline app/post.go:968]: nit: `sanitizeOptions` would be a clearer variable name

**Pull request merged**